### PR TITLE
Habits refactoring 1273

### PR DESCRIPTION
**Pull request by SashkoMolodec**
Need to rewrite tests for several classes.

### CODE REVIEW

**Comment by antonbakalets** [comment]
There is no reason to review as this branch is far beyond dev. Please merge dev first.

**Review by antonbakalets — COMMENTED** [review]
I don't like commented out code, but you promised to fix it.

**Review by Yura09 — APPROVED** [review]

File: core/src/main/java/greencity/repository/HabitAssignRepo.java [hunk 1]
```diff
@@ -123,11 +123,10 @@ int countMarkedHabitAssignsByUserIdAndPeriod(@Param("userId") Long userId,
     /**
      * Method to delete {@link HabitAssign} by it's id.
      *
+     *
      * @param id {@link HabitAssign} id.
      */
-    @Modifying
-    @Query("DELETE FROM HabitAssign ha WHERE ha.id = :id")
-    void deleteById(@Param("id") Long id);
+    void deleteById(Long id);
```
Comment (antonbakalets) [inline core/src/main/java/greencity/repository/HabitAssignRepo.java:129]: I'm not sure about Modifying.

**Review by antonbakalets — APPROVED** [review]

**Review by forestik — APPROVED** [review]

File: core/src/main/java/greencity/repository/HabitFactTranslationRepo.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                                                                                  String habitFact);
+
+    /**
+     * Method finds random {@link HabitFact} in 3 languages between all facts that were not used during this iteration.
+     *
+     * @return optional list of {@link HabitFactTranslation}.
+     */
+    @Query(nativeQuery = true, value = "SELECT * FROM habit_fact_translations where habit_fact_id in "
+        + "(select habit_fact_id from habit_fact_translations where fact_of_day_status = 0 order by RANDOM() LIMIT 1)")
+    Optional<List<HabitFactTranslation>> findRandomHabitFact();
```
[the remark this answers is not in the document]
Comment (SashkoMolodec) [reply core/src/main/java/greencity/repository/HabitFactTranslationRepo.java]: Changes done.

File: core/src/main/java/greencity/dto/advice/AdviceDto.java [hunk 3]
```diff
@@ -1,28 +1,25 @@
...
+import greencity.dto.habit.HabitDto;
 import javax.validation.constraints.Min;
 import javax.validation.constraints.NotBlank;
-import lombok.AllArgsConstructor;
-import lombok.EqualsAndHashCode;
-import lombok.Getter;
-import lombok.NoArgsConstructor;
-import lombok.Setter;
+import lombok.*;
 import org.hibernate.validator.constraints.Length;
```
[the remark this answers is not in the document]
Comment (SashkoMolodec) [reply core/src/main/java/greencity/dto/advice/AdviceDto.java]: Changes done.

File: core/src/main/java/greencity/repository/HabitAssignRepo.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+     * @param userId {@link User} id.
+     * @param start  {@link ZonedDateTime} start time.
+     * @param end    {@link ZonedDateTime} end time.
+     * @return amount of items in Optional in case of absence such info.
+     */
+    @Query(value = "SELECT COUNT(ha) "
+        + "FROM HabitAssign ha "
+        + "WHERE ha.suspended = false "
+        + "AND ha.user.id = :userId "
+        + "AND ha.createDate > :start OR ha.createDate < :end")
```
[the remark this answers is not in the document]
Comment (SashkoMolodec) [reply core/src/main/java/greencity/repository/HabitAssignRepo.java]: Changes done.

File: core/src/main/java/greencity/repository/HabitAssignRepo.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        + "SET ha.suspended = true WHERE ha.id = :habitAssignId")
+    void suspendById(@Param("habitAssignId") Long habitAssignId);
+
+    /**
+     * Method to delete {@link HabitAssign} by it's id.
+     *
+     * @param id {@link HabitAssign} id.
+     */
+    @Modifying
+    @Query("DELETE FROM HabitAssign ha WHERE ha.id = :id")
```
[the remark this answers is not in the document]
Comment (SashkoMolodec) [reply core/src/main/java/greencity/repository/HabitAssignRepo.java]: Changes done.

File: core/src/main/java/greencity/repository/HabitAssignRepo.java (hunk 1 above)
Comment (SashkoMolodec) [reply core/src/main/java/greencity/repository/HabitAssignRepo.java:129]: Works without Modifying.

**Pull request merged**